blog/_posts/2018-02-14-tooling.md

```diff
@@ -0,0 +1,187 @@
+---
+layout: blog-detail
+post-type: blog
+by: Jon Pretty
+title: "Towards A Brighter Tooling Future for Scala"
+---
+
+# Towards A Brighter Tooling Future for Scala
+
+Scala has, for its entire existence, been on the back foot with its tooling
+support. This was true when the language first arrived in 2004 as little more
+than a command-line compiler, through 2010 when Typesafe's (now Lightbend's)
+push for commercial adoption took an early focus on improving the Scala IDE for
+Eclipse, through to today, as the tooling experience is very much improved
+thanks in part to the work of JetBrains on IntelliJ, but still lags behind the
+tooling support some other languages enjoy.
+
+People care about language tools, and it says a lot for Scala that its success
+has come without ever having had the best tooling. But a better development
+experience in other languages has always made their grass seem greener, and
+over the years has undoubtedly limited the adoption of Scala.
+
+Now, in 2018, there have been some developments primarily outside of the Scala
+community which present an new opportunity to change that.
+
+## The Language Server Protocol
+
+The language server protocol (LSP) initiative, started by Microsoft, but
+already seeing wide adoption, is a language-agnostic protocol for allowing
+tools which consume information about code ("clients", such as IDEs) to
+communicate with compilers and other tools which can provide that information
+("servers", or as the protocol describes them, "language smartness providers").
+The LSP proposal has had early successes and has collected a lot of momentum
+behind it. So it's no surprise that there's already a lot of enthusiasm within
+the Scala community around the idea.
+
+During discussions in the last Scala Center Advisory Board meeting, one of the
+Scala Center sponsors proposed that the Scala Center could support the
+initiative, so we scheduled a meeting in January to get the ball rolling.
+
+With a many players from both commercial and open-source backgrounds already
+involved in tool development for Scala, we wanted to make sure that all the
+developers who had already contributed significantly to Scala's tooling could
+attend, but without having so many attendees as to make the meeting
+unmanageable. So we privately invited about fifty people, mostly suggested by
+the Advisory Board members.
+
+This "closed doors" approach for the initial meeting was a practical
+compromise, made just for this one-off meeting. There are certainly other
+people in the community who have contributed to tooling who didn't get the
+chance to attend. To those we overlooked: sorry! we certainly want to include
+everyone who cares deeply about Scala's tooling, but it wasn't possible to run
+a meeting with more than about thirty attendees.
+
+## The Initial Meeting
+
+The goals of this first meeting were, nevertheless, modest.
```

Comment:
Nevertheless?

Reply:
The intention was, "nevertheless [despite the many people there]".

Reply:
Oh, I didnt catch that.